codebuild/bin/install_openssl_3_0.sh

```diff
@@ -17,27 +17,44 @@ set -ex
 pushd "$(pwd)"
 
 usage() {
-    echo "install_openssl_3_0.sh build_dir install_dir os_name"
+    echo "install_openssl_3_0.sh build_dir install_dir os_name [fips]"
     exit 1
 }
 
-if [ "$#" -ne "3" ]; then
+if [ "$#" -eq "3" ]; then
+    FIPS=false
+elif [ "$#" -eq "4" ] && [ "$4" = "fips" ]; then
+    FIPS=true
+else
     usage
 fi
 
 BUILD_DIR=$1
 INSTALL_DIR=$2
 OS_NAME=$3
 source codebuild/bin/jobs.sh
-RELEASE=3.0.7
+prelude=$(cat codebuild/bin/openssl_fips_prelude)
+
+# Only some versions of Openssl-3 are FIPS validated.
+# The list can be found at https://openssl-library.org/source/
+if $FIPS; then
+    RELEASE=3.0.9
+else
+    RELEASE=3.0.7
```

Comment:
Maybe a note on why we need to stay on 3.0.7 for non-Fips ?

Reply:
We don't HAVE to, my thinking is just that the two should be independent. FIPS will always be limited, and non-FIPS will always be update-able. So might as well have two separate versions from the beginning.